include/seqan3/io/detail/output_iterator_conversion_adaptor.hpp

```diff
@@ -0,0 +1,217 @@
+    {}
+    output_iterator_conversion_adaptor & operator= (output_iterator_conversion_adaptor const &) = default;
+    output_iterator_conversion_adaptor & operator= (output_iterator_conversion_adaptor &&) = default;
+    ~output_iterator_conversion_adaptor() = default;
+    //!\}
+
+    /*!\name Member functions
+     * \{
+     */
+    //!\brief Inserts an object into the associated writable object.
+    constexpr output_iterator_conversion_adaptor & operator= (alpha_type const c)
+    {
+        *oiter = c;
+        ++oiter;
+        return *this;
+    }
+
+    //!\brief Inserts an object into the associated writable object from char.
+    constexpr output_iterator_conversion_adaptor & operator= (char const c)
+    //!\cond
+        requires !same_concept<alpha_type, char>
+    //!\endcond
+    {  // Explicit conversion through char conversion.
+        *oiter = assign_char(alpha_type{}, c);
+        ++oiter;
+        return *this;
+    }
+
+    //!\brief no-op
+    constexpr output_iterator_conversion_adaptor & operator* ()
+    {
+        return *this;
+    }
+
+    //!\brief no-op
+    constexpr output_iterator_conversion_adaptor & operator++ ()
+    {
+        return *this;
+    }
+
+    //!\brief no-op
+    constexpr output_iterator_conversion_adaptor & operator++ (int)
+    {
+        return *this;
+    }
+    //!\}
+};
+
+/*!\name Convenience functions
+ * \{
+ */
+/*!\brief A convenience function template that constructs a seqan3::detail::output_iterator_conversion_adaptor for the
+ * container `c` with the type deduced from the type of the argument.
+ * \tparam container_type Any type that satisfies the seqan3::sequence_concept.
+ * \param c The instance to construct a push back iterator for.
+ * \returns seqan3::detail::output_iterator_conversion_adaptor over the output container.
+ * \ingroup io
+ * \relates output_iterator_conversion_adaptor
+ */
+template <typename container_type>
+//!\cond
+    requires sequence_concept<std::remove_reference_t<container_type>> &&
+             !std::is_const_v<std::remove_reference_t<container_type>>
+//!\endcond
+auto make_conversion_output_iterator(container_type & c)
+{
+    using value_type  = typename container_type::value_type;
+    using insert_iter = ranges::back_insert_iterator<container_type>;
+    using out_iter    = output_iterator_conversion_adaptor<insert_iter, value_type>;
+
+    return out_iter{insert_iter{c}};
+}
+
+/*!\brief A convenience function template that constructs an seqan3::detail::output_iterator_conversion_adaptor for the
+ * ostream `s` with the char type and traits type deduced from the type of the argument.
+ * \tparam ostream_type Any type that satisfies the seqan3::ostream_concept.
+ * \param s The stream instance to construct the ostreambuf_iterator for.
+ * \returns seqan3::detail::output_iterator_conversion_adaptor over the output stream.
+ * \ingroup io
+ * \relates output_iterator_conversion_adaptor
+ */
```

Comment:
Adding some indention to your dox would really improve readability, e.g.

```
+/*!\brief A convenience function template that constructs a seqan3::detail::output_iterator_conversion_adaptor for the
+ * container `c` with the type deduced from the type of the argument.
+ * \tparam container_type Any type that satisfies the seqan3::sequence_concept.
+ * \param c The instance to construct a push back iterator for.
+ * \returns seqan3::detail::output_iterator_conversion_adaptor over the output container.
+ * \ingroup io
+ * \relates output_iterator_conversion_adaptor
```

vs
```
/*!\brief      A convenience function template that constructs a seqan3::detail::output_iterator_conversion_adaptor for the
 * container `c` with the type deduced from the type of the argument.
 * \tparam     container_type Any type that satisfies the seqan3::sequence_concept.
 * \param      c The instance to construct a push back iterator for.
 * \returns    seqan3::detail::output_iterator_conversion_adaptor over the output container.
 * \ingroup    io
 * \relates    output_iterator_conversion_adaptor
```

Also please make all your `\param` either `\param[in]` or `\param[out]`